### PR TITLE
Download to memory

### PR DESCRIPTION
**Pull request by kyleishie**
Implemented a download method that returns Data instead of writing to local file path.  This is useful when disk isn't needed or available.

### CODE REVIEW

File: Sources/Shout/SFTP.swift [hunk 1]
```diff
@@ -107,6 +107,40 @@ public class SFTP {
...
+        
+        var downloadedData = Data()
+        
+        var dataLeft = true
+        while dataLeft {
+            switch sftpHandle.read() {
+            case .data(let data):
+                downloadedData.append(data)
+            case .done:
+                dataLeft = false
```
Comment (beefon) [inline Sources/Shout/SFTP.swift:133]: can you replace `var dataLeft` with `while true` and `break` here?

Comment (beefon) [reply Sources/Shout/SFTP.swift:133]: perhaps. :)

**Comment by jakeheis** [comment]
Looks good to me! Could you rebase this off master? Would like to make sure CI build passes before merging

File: Sources/Shout/SFTP.swift (hunk 1 above)
Comment (kyleishie) [reply Sources/Shout/SFTP.swift:133]: Thats what I get for copying others code. 😅

**Comment by kyleishie** [comment]
I had to recreate the branch in order to get around an error.  It's back up now.  Do you want to reopen this PR or should I resubmit?

**Comment by jakeheis** [comment]
You can go ahead and submit a PR for the new branch

**Pull request closed without merging**